modules/ROOT/pages/netsuite/netsuite-examples.adoc

```diff
@@ -0,0 +1,560 @@
+. Change the display name of the *Search* operation to `Search Customer Advanced`.
+. Select an existing global element or create a new one for the *Search* operation.
+. On the properties window:
+** In the *Key* field, select `CUSTOMER_ADVANCED`.
+** Set the *Page Size* field value to an integer value between `10` and `1000`.
+
+=== Add the First Transform Message Component
+
+This Transform Message Component configures custom labels for the search results and specifies the search criteria for the flow:
+
+. From the *Mule Palette* view, drag the *Transform Message* component to the left of *Search*.
+. Change the name of the *Transform Message* component to `Search Criteria`.
+. Click the *Transform Message* component.
++
+The *Output* column displays the metadata of the *CustomerSearchAdvanced* object.
++
+. To configure the fields in the search results:
+.. Under *CustomerSearchAdvanced*, expand the *basic* object.
+.. Expand *dateCreated*.
+.. Double-click the *customLabel* field and set its value to `"Created"`.
+.. Under the *entityId* object, double-click the *customLabel* field and set its value to `"Entity ID"`.
+.. Under the *stage* object, double-click the *customLabel* field and set its value to `Stage`.
+. To configure the search criteria for the flow:
+.. Under *CustomerSearchAdvanced*, expand the *Criteria* object.
+.. Expand the *basic* object.
+.. Under the *dateCreated* object, double-click the *operator* field and set its value to `"AFTER"`.
+.. Double-click the *searchValue* field and set its value to `attributes.queryParams["dateCreated"] as LocalDateTime {format: "yyyy-MM-dd HH:mm:ss.SSS"}`.
++
+This setting converts the value of the `dateCreated` query parameter to the `LocalDateTime` format.
++
+.. Expand the *stage* object.
+.. Double-click the *operator* field and set its value to `"ANY_OF"`.
+.. Double-click the *searchValue* field and set its value to `[attributes.queryParams["stage"]]`.
++
+This setting converts the value from the `stage` query parameter to an array item.
+
+The DataWeave code should look like this:
+
+[source,dataweave,linenums]
+----
+%dw 2.0
+output application/java
+---
+{
+	columns: {
+		basic: {
+			dateCreated: [{
+				customLabel: "Created"
+			}],
+			entityId: [{
+				customLabel: "Entity Id"
+			}],
+			stage: [{
+				customLabel: "Stage"
+			}]
+		}
+	},
+	criteria: {
+		basic: {
+			dateCreated: {
+				operator: "AFTER",
+				searchValue: attributes.queryParams["dateCreated"] as LocalDateTime {format: "yyyy-MM-dd HH:mm:ss.SSS"}
+			},
+			stage: {
+				operator: "ANY_OF",
+				searchValue: [attributes.queryParams["stage"]]
+			}
+		}
+	}
+} as Object {
+	class : "org.mule.module.netsuite.extension.api.CustomerSearchAdvanced"
+}
+----
+
+=== Add a For Each component
+
+The For Each component processes each records on the list returned by the Search operation individually so that the records can be displayed on the console.
```

Comment:
```suggestion
The For Each component processes each record on the list returned by the Search operation individually so that the records can be displayed on the console.
```
OR
```suggestion
The For Each component processes each of the records on the list returned by the Search operation individually so that the records can be displayed on the console.
```